src/osd/OSDMap.cc

```diff
@@ -5784,3 +5843,20 @@ float OSDMap::pool_raw_used_rate(int64_t poolid) const
     ceph_abort_msg("unrecognized pool type");
   }
 }
+
+unsigned OSDMap::get_crush_node_flags(int osd) const
+{
+  unsigned flags = 0;
+  if (!crush_node_flags.empty()) {
+    // the map will contain type -> name
+    std::map<std::string,std::string> ploc = crush->get_full_location(osd);
```

Comment:
What if we have different flags for different parents? E.g.,
```
rack-1              {no-out, no-down}
host-1              {no-out, no-in}
```
or we should not allow that to happen at all..

Reply:
in that case, if host-1 is in rack-1, an osd on host-1 would have all 3 flags set...